### PR TITLE
Fix panic when resource's owning interface is not otherwise implicitly imported

### PR DESCRIPTION
**Pull request by rylev**
Fixes https://github.com/bytecodealliance/wac/issues/99

The issue is that when the aggregator remaps resources, it does not keep track of whether the resource's owning interface needs to be implicitly imported or not. The tracking for implicitly imported types is kept in the [`names`](https://github.com/bytecodealliance/wac/blob/ac04147f33a6e055c986023b8d600c49e942e0c1/crates/wac-types/src/aggregator.rs#L26) collection. After [remapping a type, it is put into the `names` collection](https://github.com/bytecodealliance/wac/blob/ac04147f33a6e055c986023b8d600c49e942e0c1/crates/wac-types/src/aggregator.rs#L68-L69), and those imports are then [encoded in `wac_graph` as imports](https://github.com/bytecodealliance/wac/blob/ac04147f33a6e055c986023b8d600c49e942e0c1/crates/wac-graph/src/graph.rs#L1536-L1540).

If the owning interface is not otherwise implicitly or explicitly imported, there will be a panic because we assume the interface has been seen when we go to encode.

### CODE REVIEW

File: crates/wac-graph/tests/graphs/implicit-resource-import/encoded.wat [hunk 1]
```diff
@@ -0,0 +1,57 @@
+(component
+  (type (;0;)
+    (instance
+      (export (;0;) "x" (type (sub resource)))
+    )
+  )
+  (import "foo:dependency/types" (instance (;0;) (type 0)))
+  (alias export 0 "x" (type (;1;)))
+  (import "x" (type (;2;) (eq 1)))
```
Comment (rylev) [inline crates/wac-graph/tests/graphs/implicit-resource-import/encoded.wat:9]: I'm not sure if this is correct...

Comment (peterhuene) [reply crates/wac-graph/tests/graphs/implicit-resource-import/encoded.wat:9]: This is correct; a `use` of a resource in a world is an import of that resource, not an import of the interface and then aliasing the export of that interface.

File: crates/wac-graph/tests/encoding.rs [hunk 2]
```diff
@@ -331,6 +332,7 @@ fn encoding() -> Result<()> {
             })?
             .into_composition_graph(&path, test_case)
             .and_then(|graph| {
+                println!("{:?}", graph);
```
Comment (peterhuene) [inline crates/wac-graph/tests/encoding.rs:335]: Debugging `println!`?

**Review by peterhuene — COMMENTED** [review]
Looks great, just the one comment!

File: crates/wac-graph/tests/encoding.rs (hunk 2 above)
Comment (rylev) [reply crates/wac-graph/tests/encoding.rs:335]: This has been useful for me. It should only show up if you pass `--nocapture` to cargo test. Happy to remove it if you find it distracting though.

Comment (peterhuene) [reply crates/wac-graph/tests/encoding.rs:335]: We can keep it in and remove it later should it prove too noisy.

**Review by peterhuene — APPROVED** [review]

**Pull request merged**